### ISSUE TITLE
Filtering chunks

### ISSUE THREAD

**Issue by mooyoul** [GoogleChrome/preload-webpack-plugin#6]
It would be nice if preload-webpack-plugin supports filtering chunks (only include some chunks/exclude some chunks.) [like html-webpack-plugin's one](https://github.com/jantimon/html-webpack-plugin#filtering-chunks).

It will be useful on some use-cases.
1. Some chunks (e.g. Admin page) may contain some sensitive information. so user wants to exclude it from preload targets. (blacklisting)
2. Some chunks don't need to be preloaded. so user wants to include some chunks only (whitelisting)

**Comment by addyosmani** [comment]
I think that's a valid ask. You're proposing using a similar API to how html-webpack-plugin tackles this I assume? :)

**Comment by mooyoul** [comment]
Yeah that's correct. I just want to include/exclude some chunks on preload-webpack-plugin like html-webpack-plugin :)

FYI, previously i was used below snippet (quick & dirty solution) for implementing preload chunks with filtering on my html-webpack-plugin html template file:

```html
    <%
      Object.keys(webpack.assetsByChunkName)
        .filter((k) => k && k.indexOf('CHUNK_NAME_TO_BE_IGNORED') === -1)
        .reduce((collection, k) => collection.concat(webpack.assetsByChunkName[k]), [])
        .filter((filename) => filename && filename.slice(-3) === '.js')
        .forEach((filename) => {
    %>
    <link rel="preload" href="<%= webpackConfig.output.publicPath %><%= filename %>" as="script">
    <%  }); %>
```

**Comment by jantimon** [comment]
The best way would be to use the chunks from `htmlPluginData` which gives you the setting per generated page as you can use multiple html-webpack-plugin instances https://github.com/GoogleChrome/preload-webpack-plugin/blob/7760ef791605c05cd306c0f63ce9b6d6318a2c7c/index.js#L36

**Comment by ingro** [comment]
This option would be very useful! I wanted to exclude sourcemap's chunks but it seems not possible at the moment...

**Comment by jackfranklin** [comment]
This would be useful for avoiding sourcemaps - maybe you could pass in a blacklist of regexes/globs to match against? It could be populated with `['*.map']` for a start to stop the plugin including sourcemaps.

**Comment by jackfranklin** [comment]
@addyosmani I'm happy to attempt a PR if you're happy with that approach, or to talk through it.

**Comment by limichange** [comment]
> This would be useful for avoiding sourcemaps - maybe you could pass in a blacklist of regexes/globs to match against? It could be populated with ['*.map'] for a start to stop the plugin including sourcemaps.

This is useful. We don't want to preload sourcemap.

**Comment by jackfranklin** [comment]
I opened a PR with the blacklist feature: https://github.com/GoogleChrome/preload-webpack-plugin/pull/16

If anyone needs it for now you can always depend on my version of this repo.
